cpp/tests/groupby/lists_tests.cpp

```diff
@@ -64,70 +63,18 @@ static constexpr auto null = -1;
 // with LISTS keys.
 auto sum_agg() { return cudf::make_sum_aggregation<groupby_aggregation>(); }
 
-// TODO: this is a naive way to compare expected key/value against resulting key/value. To be
-// replaced once list lex comparator is supported (https://github.com/rapidsai/cudf/issues/5890)
-template <typename Equal>
-struct match_expected_fn {
-  match_expected_fn(cudf::size_type const num_rows, Equal equal)
-    : _num_rows{num_rows}, _equal{equal}
-  {
-  }
-
-  __device__ bool operator()(cudf::size_type const idx)
-  {
-    for (auto i = _num_rows; i < 2 * _num_rows; i++) {
-      if (_equal(idx, i)) { return true; }
-    }
-    return false;
-  }
-
-  cudf::size_type const _num_rows;
-  Equal _equal;
-};
-
 inline void test_hash_based_sum_agg(column_view const& keys,
```

Comment:
`structs_tests.cpp` also has this function `test_hash_based_sum_agg`. Can we unify them? or simply replace with `test_single_agg` itself.

Reply:
I think the previous comment thread resolved on not replacing those, but I can do so in `lists_tests.cpp` and `struct_tests.cpp`.